File: Algorithmen/3x+1/py/3x+1.py
import time
import math

#boilerplate
max_n = 0
max_count = 0

n = int(input("Enter a number:"))

start_time = time.time()

#calculate
count = 0
i = n
while i != 1:
    if i % 2 == 0:
        i = i / 2
    else:
        i = 3 * i + 1
    count += 1

if count > max_count:
    max_count = count
    max_n = n

#print results
elapsed_time = math.floor((time.time()-start_time)*1000)

print("Starting number:", n)
print("Number of iterations:", max_count)
print("Execution time:", elapsed_time, "ms")

File: Algorithmen/3x+1/cpp/3x+1.cpp
#include <iostream>
#include <ctime>
using namespace std;

int main() {
    //boilerplate
    int_fast64_t max_n = 0;
    int max_count = 0;
    int start_time = clock();

    //prompt user for input
    int_fast64_t n;
    cout << "Enter a number: ";
    cin >> n;

    //calculate
    int count = 0;
    int_fast64_t i = n;
    while (i != 1) {
        if (i % 2 == 0) {
            i = i / 2;
        } else {
            i = 3 * i + 1;
        }
        count++;
    }

    //print result
    cout << "Starting number: " << n << endl;
    cout << "Number of iterations: " << count << endl;
    cout << "Execution time: " << (clock() - start_time) << "ms" << endl;
}
